GUIDELINES.md

```diff
@@ -171,6 +171,28 @@ Example:
 }
 ```
 
+### Request Nullable/Optional properties
+
+#### When to use `null` in properties
+
+1. When you want to unassign the value of that property.
+2. If the property is optional and you are creating it you might want to use `null` to avoid the `if`s burden.
+
+#### When not to use `null` in properties
+
+1. Whenever you are updating a property that is optional/nullable and has a value assigned and you don't want to remove that value.
+
+#### Nullable vs Optional
+
+When you set a property to `null` your intention is to remove any assigned value to it.
+When a property is optional, you don't need to send the key in your request, and the value will be unchanged.
```

Comment:
IIRC this is different for custom fields. When updating an entity with the `custom_fields` key, all the missing optional custom fields (other than auto increment ones) will be cleared. So you are obliged to pass the current values if you want to keep them. 

Although the custom field values are not defined as properties of the entity, I still think we might have to note this in the documentation here.

Reply:
It's basically like that when talking about collections, not only custom fields (f.e. telephones, addresses, invoice items,...)

Reply:
But a collection is the value not the key of the property, right? But probably would make sense to explicit say that in the guidelines. When updating collections if you don't provide all items that already exist they will be removed. What do you think?

Reply:
That kinda makes sense.